src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -92,18 +95,45 @@ public HttpConnectionInterceptorContext interceptRequest(HttpConnectionIntercept
 
     @Override
     public HttpConnectionInterceptorContext interceptResponse(HttpConnectionInterceptorContext
-                                                                          context) {
+                                                                      context) {
         HttpURLConnection connection = context.connection.getConnection();
         try {
-            if (context.connection.getConnection().getResponseCode() == 401) {
-                //we need to get a new cookie
-                cookie = getCookie(connection.getURL());
-                //don't resend request, failed to get cookie
-                if (cookie != null) {
-                    context.replayRequest = true;
-                } else {
-                    context.replayRequest = false;
-                }
+            int statusCode = connection.getResponseCode();
+            switch (statusCode) {
+                case HttpURLConnection.HTTP_FORBIDDEN: //403
+                    //check if it was an expiry case
+                    InputStream errorStream = connection.getErrorStream();
+                    String errorString = new String(IOUtils.toString(errorStream, "UTF-8"));
+                    try {
+                        JsonObject errorResponse = new Gson().fromJson(errorString, JsonObject
```

Comment:
We can only read the error stream once and we need the content for either the exception or to know the cookie is expired, but we don't know which until we've read the stream. The error stream might not always be JSON so I thought it made sense just to use a string.